core/src/test/scala/org/apache/spark/sql/delta/ShowTableColumnsSuite.scala

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import org.apache.spark.sql.delta.test.DeltaSQLCommandTest
+import org.apache.spark.sql.functions.struct
+import org.apache.spark.sql.{AnalysisException, DataFrame, QueryTest, Row}
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.util.Utils
+
+import java.io.File
+
+trait ShowTableColumnsSuiteBase
+    extends QueryTest
+    with SharedSparkSession
+    with DeltaSQLCommandTest
+    with DeltaTestUtilsForTempViews {
+
+  import testImplicits._
+
+  private val outputColumnNames = Seq("columnName")
+  private val outputColumnValues = Seq(Seq("column1"), Seq("column2"))
+
+  protected def checkResult(
+      result: DataFrame,
+      expected: Seq[Seq[Any]],
+      columns: Seq[String]
+  ): Unit = {
+    checkAnswer(
+      result.select(columns.head, columns.tail: _*),
+      expected.map { x =>
+        Row(x: _*)
+      }
+    )
+  }
+
+  private def showDeltaColumnsTest(
+      f: File => String,
+      schemaName: String,
+      tableFormat: String
+  ): Unit = {
+    withDatabase("delta") {
+      val tempDir = Utils.createTempDir()
+      Seq(1 -> 1)
+        .toDF("column1", "column2")
+        .write
+        .format(tableFormat)
+        .mode("overwrite")
+        .save(tempDir.toString)
+
+      val sqlCommand = if (schemaName.nonEmpty) {
+        s"SHOW COLUMNS IN ${f(tempDir)} FROM $schemaName"
+      } else {
+        s"SHOW COLUMNS FROM ${f(tempDir)}"
+      }
+
+      checkResult(
+        sql(sqlCommand),
+        outputColumnValues,
+        outputColumnNames
+      )
+    }
+  }
+
+  private def showNonDeltaColumnsTest(tableFormat: String): Unit = {
+    withTable("show_columns") {
+      sql(s"""
+          |CREATE TABLE show_columns(column1 INT, column2 INT)
+          |USING $tableFormat
+          |COMMENT "describe a non delta table"
+          """.stripMargin)
+      checkResult(
+        sql("SHOW COLUMNS IN show_columns"),
+        outputColumnValues,
+        outputColumnNames
+      )
+    }
+  }
+
+  test("delta table: path") {
+    showDeltaColumnsTest(f => s"'${f.toString}'", "", "delta")
```

Comment:
nit: for constants passed to function arguments: have `parameterName = constant`. Ex. `tableName = "delta"`

Reply:
Fixed.